cfme/configure/configuration/server_settings.py

```diff
@@ -0,0 +1,272 @@
+    def is_displayed(self):
+        return (
+            self.authentication_mode.is_displayed and
+            self.authentication_mode.selected_option == 'LDAP'
+        )
+
+
+class LdapsAuthenticationView(LdapAuthenticationView):
+    """ Ldaps Authentication View """
+
+    @property
+    def is_displayed(self):
+        return (
+            self.authentication_mode.is_displayed and
+            self.authentication_mode.selected_option == 'LDAPS'
+        )
+
+
+class AmazonAuthenticationView(ServerAuthenticationView):
+    """ Amazon Authentication View """
+
+    access_key = Input(name='authentication_amazon_key')
+    secret_key = Input(name='authentication_amazon_secret')
+
+    get_groups = Checkbox(name='amazon_role')
+    validate = Button('Validate')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.authentication_mode.is_displayed and
+            self.authentication_mode.selected_option == 'Amazon'
+        )
+
+
+class ExternalAuthenticationView(ServerAuthenticationView):
+    """ External Authentication View """
+
+    enable_sso = Checkbox(name='sso_enabled')
+    enable_saml = Checkbox(name='saml_enabled')
+
+    get_groups = Checkbox(name='httpd_role')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.authentication_mode.is_displayed and
+            self.authentication_mode.selected_option == 'External (httpd)'
+        )
+
+
+class AuthenticationSetting(Navigatable, Updateable, Pretty):
+    """
+        Represents Authentication Setting for CFME
+
+        Args:
+            auth_mode: authorization mode, default value 'Database'
+    """
+
+    pretty_attrs = ['auth_mode']
+
+    user_type_dict = {
+        'userprincipalname': 'User Principal Name',
+        'dn-uid': 'Distinguished Name (UID=<user>)'
+    }
+
+    def __init__(self, auth_mode='Database', appliance=None):
+        self.auth_mode = auth_mode.capitalize()
+        Navigatable.__init__(self, appliance=appliance)
+
+    def set_session_timeout(self, hours=None, minutes=None):
+        """
+            Sets the session timeout of the appliance.
+            Args:
+                hours(str): timeout hours value
+                minutes(str): timeout minutes value
+            ex. auth_settings.set_session_timeout('0', '30')
+        """
+        view = navigate_to(self, 'Details')
+        updated = view.fill({
+            "hours_timeout": hours,
+            "minutes_timeout": minutes
+        })
+        if updated:
+            view.save_button.click()
+            flash_message = (
+                'Authentication settings saved for {} Server "{} [{}]" in Zone "{}"'.format(
+                    self.appliance.product_name,
+                    self.appliance.server_name(),
+                    self.appliance.server_id(),
+                    self.appliance.server.zone.name))
+            view.flash.assert_message(flash_message)
+
+    def update_form(self, updates, reset=False):
```

Comment:
**REQUIRED:** Do we expect people to use this function?
If not we should consider renaming it to ```_update_form```

Reply:
Done